pandas/core/generic.py

```diff
@@ -9874,7 +9874,10 @@ def ranker(data):
             data = self._get_numeric_data()
         else:
             data = self
-
+        if method != "average" and not pct:
+            temp = ranker(data)
+            temp = temp.convert_dtypes()
+            return temp
         return ranker(data)
```

Comment:
This is unnecessarily duplicating code - both paths have `ranker(data)`. Also, avoid adding more `return` statements unless it helps the structure of the code.

Reply:
Similar remarks here about the use of `convert_dtypes`.